librabbitmq/amqp_openssl_hostname_validation.c

```diff
@@ -31,25 +31,28 @@
 #include <openssl/x509v3.h>
 
 #include "amqp_hostcheck.h"
+#include "amqp_openssl_bio.h"
 #include "amqp_openssl_hostname_validation.h"
 
+#include <string.h>
+
 #define HOSTNAME_MAX_SIZE 255
 
 /**
-* Tries to find a match for hostname in the certificate's Common Name field.
-*
-* Returns AMQP_HVR_MATCH_FOUND if a match was found.
-* Returns AMQP_HVR_MATCH_NOT_FOUND if no matches were found.
-* Returns AMQP_HVR_MALFORMED_CERTIFICATE if the Common Name had a NUL character
-* embedded in it.
-* Returns AMQP_HVR_ERROR if the Common Name could not be extracted.
-*/
+ * Tries to find a match for hostname in the certificate's Common Name field.
+ *
+ * Returns AMQP_HVR_MATCH_FOUND if a match was found.
+ * Returns AMQP_HVR_MATCH_NOT_FOUND if no matches were found.
+ * Returns AMQP_HVR_MALFORMED_CERTIFICATE if the Common Name had a NUL character
+ * embedded in it.
+ * Returns AMQP_HVR_ERROR if the Common Name could not be extracted.
+ */
 static amqp_hostname_validation_result amqp_matches_common_name(
     const char *hostname, const X509 *server_cert) {
   int common_name_loc = -1;
   X509_NAME_ENTRY *common_name_entry = NULL;
   ASN1_STRING *common_name_asn1 = NULL;
-  char *common_name_str = NULL;
+  const char *common_name_str = NULL;
```

Comment:
Instead of making change to these functions, use an `#ifdef` and use [`X509_check_host`](https://www.openssl.org/docs/manmaster/man3/X509_check_host.html) to do the hostname validation.

Reply:
The docs of X509_check_host say: "Applications are encouraged to use X509_VERIFY_PARAM_set1_host() rather than explicitly calling X509_check_host. " and there is some documentation on https://wiki.openssl.org/index.php/Hostname_validation on how to use it .

X509_check_host is not even a new function in 1.1.0, it was added in 1.0.2. 

I don't want to say that this does not make sense, but can't it be done separately from this change?